job_progress/job_progress.py

```diff
@@ -120,17 +120,62 @@ def __exit__(self, exc_type, exc_value, traceback):
         if self.delete_on_closing:
             self.delete()
 
-    def add_one_progress_state(self, state):
+    def add_one_progress_state(self, state, item_id=None):
         """Add one unit status."""
-        return self.backend.add_one_progress_state(self.id, state)
+        return self.backend.add_one_progress_state(self.id, state, item_id)
 
-    def add_one_failure(self):
-        """Add one failure state."""
-        return self.add_one_progress_state(states.FAILURE)
+    def add_one_failure(self, item_id=None):
+        """Add one failure state.
 
-    def add_one_success(self):
-        """Add one success state."""
-        return self.add_one_progress_state(states.SUCCESS)
+        :param item_id: Details with this progress. If it is None, then no
+                        details will be added for this progress.
+        """
+        self.add_one_progress_state(states.FAILURE, item_id)
+
+    def add_one_success(self, item_id=None):
+        """Add one success state.
+
+        :param item_id: Details with this progress. If it is None, then no
+                        details will be added for this progress.
+        """
+        self.add_one_progress_state(states.SUCCESS, item_id)
+
+    def track(self, is_success, item_id=None):
+        """Check if an object is_success or not. If failed, put it in
```

Comment:
First line of a docstring should a one-liner. http://www.python.org/dev/peps/pep-0257/#multi-line-docstrings